novice-r-practice.Rmd

```diff
@@ -0,0 +1,363 @@
+# Practice
+
+We have covered a lot in the last few lessons,
+so this one presents some practice exercises to ground what we have learned
+and introduce a few more commonly-used functions.
+
+## Working with a single tidy table
+
+1. Load the tidyverse collection of libraries and the `here` library for constructing paths:
+
+```{r fake-load-libraries}
+library(tidyverse)
+library(here)
+```
+
+2. Use `here::here` to construct a path to a file and `readr::read_csv` to read that file:
+
+```{r read-survey-data}
+path = here::here("novice-r-practice", "person.csv")
+person <- readr::read_csv(path)
+person
+```
+
+*Read `survey/site.csv`.*
+
+3. Count rows and columns using `nrow` and `ncol`:
+
+```{r count-rows-and-columns}
+nrow(person)
+ncol(person)
+```
+
+*How many rows and columns are in the site data?*
+
+4. Format strings using `glue::glue`:
+
+```{r use-glue}
+print(glue::glue("person has {nrow(person)} rows and {ncol(person)} columns"))
+```
+
+*Print a nicely-formatted summary of the number of rows and columns in the site data.*
+
+5. Use `colnames` to get the names of columns and `paste` to join strings together:
+
+```{r use-colnames-and-paste}
+print(glue::glue("person columns are {paste(colnames(person), collapse = ' ')}"))
+```
+
+*Print a nicely-formatted summary of the names of the columns in the site data.*
+
+6. Use `dplyr::select` to create a new table with a subset of columns by name:
+
+```{r select-by-name}
+dplyr::select(person, family_name, personal_name)
+```
+
+*Create a table with just the latitudes and longitudes of sites.*
+
+7. Use `dplyr::filter` to create a new table with a subset of rows by values:
+
+```{r filter-with-one-condition}
+dplyr::filter(person, family_name < "M")
+```
+
+*Create a table with only sites south of -48 degrees.*
+
+8. Use the pipe operator `%>%` to combine operations:
+
+```{r select-pipe-filter}
+person %>%
+  dplyr::select(family_name, personal_name) %>%
+  dplyr::filter(family_name < "M")
+```
+
+*Create a table with only the latitudes and longitudes of sites south of -48 degrees.*
+
+9. Use `dplyr::mutate` to create a new column with calculated values and `stringr::str_length` to calculate string length:
+
+```{r mutate-name-length}
+person %>%
+  dplyr::mutate(name_length = stringr::str_length(family_name))
+```
+
+*Use the built-in function `round` to create a table with latitudes and longitudes rounded to integers.*
```

Comment:
Would students already have seen `round()`?  Or is this also an exercise in figuring out how to use a "new to you" function, e.g. with `?round`?

Reply:
Good catch - exercise modified to ask them to look at the help first.